pkgs/by-name/hy/hypercore/package.nix

```
@@ -0,0 +1,47 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  writeScript,
+}:
+
+buildNpmPackage (finalAttrs: {
+  pname = "hypercore";
+  version = "11.6.2";
+
+  src = fetchFromGitHub {
+    owner = "holepunchto";
+    repo = "hypercore";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-o4Ojja+8Yvn1fK9FszcH/wHZnuyhpHxsch0D1kb5HlE=";
+  };
+
+  npmDepsHash = "sha256-lBV5IehqP2qOe3/mJjLJt8Ty77QN8uUGc5oolFUEqck=";
+
+  dontNpmBuild = true;
+
+  passthru.updateScript = writeScript "update-hypercore" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p  common-updater-scripts curl
+
+    set -eu -o pipefail
+
+    version="$(curl -s https://api.github.com/repos/holepunchto/hypercore/tags | jq -r '.[0].name' | sed 's/^v//')"
+
+    update-source-version hypercore "$version"
+  '';
```

We don't need to write an update script ourselves since the `nix-update-script` will work fine (don't forget to add it as an input, above)

```suggestion
  passthru.updateScript = nix-update-script { };
```